services/core-web/src/components/mine/Permit/MinePermitTable.tsx

```diff
@@ -228,6 +234,11 @@ const columns: ColumnsType<MinePermitTableItem> = [
     key: "addEditButton",
     align: "right",
     render: (text, record) => {
+      const history = useHistory();
+      const { id } = useParams<{ id: string }>();
```

Comment:
sonarcloud is complaining about using hooks inside a render method that isn't a react component. Does it work to pull these properties out to the component?

I'm thinking that below could be simplified to `const permitGuid = record.permit?.permit_guid ?? record.permit?.permit_amendment_guid` but it's not important.

Reply:
Yeah, I was scratching my head a bit on that one.  It's not easy to contain that within the parent component because the columns are defined outside of it.

That said, I think I'll just move the columns into the component.  The current setup of this component kind of irks me since it defines a whole bunch of render methods above and outside of the actual component.  I'll save the full refactor for another time though.

Reply:
Seriously, when you don't actually get to the main component until line 650 of the file it's named after it's time to refactor.